### ISSUE TITLE
Layout of links in sidebar

### ISSUE THREAD

**Issue by ioquatix** [rubygems/rubygems.org#2375]
As discussed here: https://github.com/rubygems/rubygems/issues/3616

Firstly, on the sidebar:

![](https://user-images.githubusercontent.com/30030/81500552-f8c63100-9326-11ea-8b08-e50b498417c7.png)

I don't see any funding URI. Is this live yet?

Also, I see "Homepage" and "Source Code" both point to the same thing.

1/ I wonder if we should display the full URI somewhere - e.g. under the link in <small> or something.
2/ Are we adding the funding URI? Is this still WIP?
3/ Can we skip showing the "Homepage" link if it's in one of the more specific links?

Because, I feel like it's nicer to have "Source Code" and "Documentation" links.

Or maybe I should just remove the "documentation_uri" metadata.

Some guidance would be appreciated.

**Comment by sonalkr132** [comment]
> I don't see any funding URI. Is this live yet?

No. The feature has to be first released by the client and then we can update [this list](https://github.com/rubygems/rubygems.org/blob/master/app/models/links.rb#L3) on server side.

>  I wonder if we should display the full URI somewhere

I am not sure what do you mean by full URI. can you please elaborate. 

> Can we skip showing the "Homepage" link if it's in one of the more specific links?

`homepage` is tangled with *just legacy things*. We fall back to the legacy `linkset` table (previously set by a UI form) if `homepage_uri` is not present. This ensures we don't disregard data we have for users who haven't migrated to metadata URI. please submit a help ticket and we can reset `linkset.home` for you.

**Comment by ioquatix** [comment]
> I am not sure what do you mean by full URI. can you please elaborate.

Under the word "Homepage" show the actual link. I can't even remember why I thought it was a good idea. It's a bit confusing when several links go to the same place.

**Comment by sonalkr132** [comment]
> Under the word "Homepage" show the actual link

If you hover over a link, the full URL is shown in bottom left corner of most browsers.

**Comment by ioquatix** [comment]
I think the main point is:

1/ Funding link is not listed.
2/ Due to whatever situation sometimes "Homepage", "Documentation" and "Source Code" is all the same link, so I think we should only present the link once.

**Comment by etherbob** [comment]
Looks like someone took care of listing the funding link:

https://rubygems.org/gems/anycable

<img width="217" alt="Screenshot 2023-04-29 at 2 09 06 PM" src="https://user-images.githubusercontent.com/5025990/235320193-dcd98ea0-742b-42c7-b5a7-092c0fd23b0a.png">

**Comment by etherbob** [comment]
@ioquatix if you still want to do the coalescing of duplicate links I'd be happy to take a swing at that maybe with something like this in links.rb

```ruby
  def each
    return enum_for(:each) unless block_given?
    links.uniq do |short, long|
      send(long)
    end.each do |short, long|
      value = send(long)
      yield short, value if value
    end
  end
```

**Comment by ioquatix** [comment]
I think it would be an improvement to the UX.

**Comment by simi** [comment]
@ioquatix I think the original issue reported of missing funding link is resolved.
@etherbob which link title to show for given link?

**Comment by nobu** [comment]
I think, if there are duplicate entries, those names should be shown.

```ruby
  def each
    return enum_for(:each) unless block_given?
    links.group_by do |short, long|
      send(long)
    end.each do |short, links|
      value = send(long)
      yield links.map { |short, | short }.join(" / "), value if value
    end
  end
```

**Comment by ioquatix** [comment]
@nobu maybe worth opening a new issue explaining the problem. I think all my concerns were addressed. Thanks everyone!
